### ISSUE TITLE
regression: members of group docker defined in ldap can no longer use docker.

### ISSUE THREAD

**Issue by hboetes** [moby/moby#38202]
After upgrading from `18.06.1~ce~3-0~ubuntu` to `18.09.0~3-0~ubuntu`, the docker socket has changed permissions from `root:docker` to `root:root`. Members of a locally defined group `docker` still can run docker, members of a docker group defined by ldap or sss or freeipa can no longer use docker as user.

```
% docker run hello-world
docker: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.39/containers/create: dial unix /var/run/docker.sock: connect: permission denied.
[snip]
% getent group docker
docker:*:953200014:someuser,notheruser
# echo docker:x:200:someuser,notheruser >> /etc/group
# service docker restart
% docker run hello-world
Hello from Docker! [snip]
```

**Comment by hboetes** [comment]
Adding `-G docker` to startup options or adding `"group": "docker",` to daemon.json is ignored.

**Comment by pdoran** [comment]
Additionally, setting permissions on /var/run/docker.sock is reset after a service reboot.

**Comment by thaJeztah** [comment]
I suspect this is related to https://github.com/docker/for-linux/issues/476

Previously, the socket was created by systemd (which probably was able to do the lookup), but without socket activation, the socket will be created  by the `dockerd` daemon, which does not (yet) have that functionality. (This is being worked on in https://github.com/moby/moby/pull/38126)

**Comment by hboetes** [comment]
@thaJeztah ah, great! Thanks for the heads up!

**Comment by AlanVncs** [comment]
`sudo chgrp <LDAP GID> /var/run/docker.sock`

**Comment by thaJeztah** [comment]
With both https://github.com/docker/docker-ce-packaging/pull/257 and https://github.com/moby/moby/pull/38126 merged (and the first one backported to 18.09 https://github.com/docker/docker-ce-packaging/pull/258), I think this can be closed

**Comment by luckyraul** [comment]
@thaJeztah same problem
